mlx_graphs/data/batch.py

```diff
@@ -8,35 +7,38 @@
 
 class Batch(GraphData):
     """
-        Represents a batch data object describing a batch of graphs as one big (disconnected)
-        graph.
+    Represents a batch data object describing a batch of graphs as one big (disconnected)
+    graph.
 
     """
 
-    def __init__(self, node_features, edge_index, num_nodes) -> None:
+    def __init__(self, node_features, edge_index, cumsum) -> None:
         self.node_features = node_features
         self.edge_index = edge_index
-        self.cumsum_num_nodes = mx.cumsum(num_nodes)
+        self.cumsum = cumsum
 
     @classmethod
-    def from_graph_list(cls, graphs: List[GraphData]) -> Self :
+    def from_graph_list(cls, graphs: List[GraphData]) -> Self:
         """
         Constructs a :class:`mlx_graphs.batch.Batch` object from a
-        list of :class:`~mlx_graphs.data.GraphData` 
+        list of :class:`~mlx_graphs.data.GraphData`
         """
-        
+
         if not isinstance(graphs, (list, tuple)):
             graphs = list(graphs)
 
         num_nodes = mx.array([0] + [graph.num_nodes() for graph in graphs])
+        cumsum = mx.cumsum(num_nodes)
```

Comment:
I replaced your num_nodes sum with cumsum. I don't know why you used that?

Reply:
It was a mistake indeed.